pkg/controller/controller_binding_test.go

```diff
@@ -2151,3 +2190,289 @@ func TestReconcileServiceInstanceCredentialWithSecretParameters(t *testing.T) {
+	})
+
+	addGetNamespaceReaction(fakeKubeClient)
+	// existing Secret with nil controllerRef
+	addGetSecretReaction(fakeKubeClient, &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: testServiceInstanceCredentialName, Namespace: testNamespace},
+	})
+
+	sharedInformers.ServiceBrokers().Informer().GetStore().Add(getTestServiceBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+	sharedInformers.ServicePlans().Informer().GetStore().Add(getTestServicePlan())
+	sharedInformers.ServiceInstances().Informer().GetStore().Add(getTestServiceInstanceWithStatus(v1alpha1.ConditionTrue))
+
+	binding := &v1alpha1.ServiceInstanceCredential{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       testServiceInstanceCredentialName,
+			Namespace:  testNamespace,
+			Finalizers: []string{v1alpha1.FinalizerServiceCatalog},
+			Generation: 1,
+		},
+		Spec: v1alpha1.ServiceInstanceCredentialSpec{
+			ServiceInstanceRef: v1.LocalObjectReference{Name: testServiceInstanceName},
+			ExternalID:         bindingGUID,
+			SecretName:         testServiceInstanceCredentialSecretName,
+		},
+	}
+	binding.Status.CurrentOperation = v1alpha1.ServiceInstanceCredentialOperationBind
+	binding.Status.OperationStartTime = nil
+	binding.Status.OrphanMitigationInProgress = true
+
+	if err := testController.reconcileServiceInstanceCredential(binding); err != nil {
+		t.Fatalf("reconciliation should complete since the retry duration has elapsed: %v", err)
+	}
+	kubeActions := fakeKubeClient.Actions()
+	assertNumberOfActions(t, kubeActions, 1)
+	action := kubeActions[0].(clientgotesting.GetAction)
+	if e, a := "delete", action.GetVerb(); e != a {
+		t.Fatalf("Unexpected verb on action; expected %v, got %v", e, a)
+	}
+	if e, a := "secrets", action.GetResource().Resource; e != a {
+		t.Fatalf("Unexpected resource on action; expected %v, got %v", e, a)
+	}
+
+	brokerActions := fakeServiceBrokerClient.Actions()
+	assertNumberOfServiceBrokerActions(t, brokerActions, 1)
+	assertUnbind(t, brokerActions[0], &osb.UnbindRequest{
+		BindingID:  bindingGUID,
+		InstanceID: instanceGUID,
+		ServiceID:  serviceClassGUID,
+		PlanID:     planGUID,
+	})
+
+	actions := fakeCatalogClient.Actions()
+	assertNumberOfActions(t, actions, 1)
+
+	updatedServiceInstanceCredential := assertUpdateStatus(t, actions[0], binding).(*v1alpha1.ServiceInstanceCredential)
+	assertServiceInstanceCredentialCondition(t, updatedServiceInstanceCredential, v1alpha1.ServiceInstanceCredentialConditionReady, v1alpha1.ConditionFalse, "OrphanMitigationSuccessful")
+	assertServiceInstanceCredentialOrphanMitigationSet(t, updatedServiceInstanceCredential, false)
+}
+
+// TestReconcileBindingWithOrphanMitigationTimedOut tests
+// reconcileServiceInstanceCredential to ensure a binding is properly handled
+// once orphan mitigation is underway, specifically in the failure scenario of a
+// time out during orphan mitigation.
+func TestReconcileBindingWithOrphanMitigationTimedOut(t *testing.T) {
```

Comment:
"Timed out" is ambiguous here; can we rename this to be something like `TestReconcileBindingWithOrphanMitigationReconciliationRetryTimeout`